conpy/connectivity.py

```diff
@@ -1367,43 +1392,111 @@ def dics_coherence_external(csd, dics, info, external=None):
         for i in _picks_to_idx(info, ["eeg", "meg"])
     ]
     freqs = csd.frequencies
-    coherences = np.zeros((len(picks_external), dics["n_sources"], len(freqs)))
 
+    if not dics['is_free_ori']:
+        n_orient = 1
+    elif len(dics['source_nn']) // dics['n_sources'] == 2:
+        raise ValueError("Forward solution must be in free orientation. The "
+                         "provided forward solution seems to be in tangential "
+                         "orientation.")
+    elif len(dics['source_nn']) // dics['n_sources'] == 3:
+        n_orient = 3
+    else:
+        raise ValueError("Invalid source_nn in dics beamformer.")
+
+    if maximize:
```

Comment:
MNE-Python convention is to always make this sort of setting some value of `pick_ori`. Could we build it such that we have `pick_ori="max-coherence"` which performs the search, or `pick_ori="max-power"` that uses the first principal component?

Reply:
`pick_ori` of `dics` can be either `max-power`, `normal`, or `vector`. If we want `pick_ori="max-coherence"`, I guess it will need to be a separate parameter passed to `dics_coherence_external`.

> or even better, unless we can think of a compelling reason not to want the max-coherence orientation, why not skip the parameter altogether and just always do it this way?

I guess we could set it so that if `dics['is_free_ori']`, the `max-coherence` orientation is always used. This of course hides the setting, so the user would need to know that `pick_ori` in `make_dics` has big consequences here.

Reply:
How about having `pick_ori='max-coherence`` as the default. Then throw an error if the beamformer is not in free orientation. Then if the user really wants they can set `pick_ori='max-power'` or `'pick_ori='normal', which also requires that the beamformer was calculated accordingly.